src/core/mesh/qgsmeshdataprovider.h

```diff
@@ -475,6 +475,17 @@ class CORE_EXPORT QgsMeshDataProvider: public QgsDataProvider, public QgsMeshDat
      */
     virtual void close() = 0;
 
+    /**
+     * \brief Remove dataset group from the mesh
+     *
+     * emits dataChanged when successful
+     *
+     * \return True on success
+     *
+     * \since QGIS 3.42
+     */
+    virtual bool removeDatasetGroup( const int &index ) = 0;
```

Comment:
We pass trivial types by value instead of const reference:

```suggestion
    virtual bool removeDatasetGroup( int index ) = 0;
```
(also in other places)